CRM/Core/DAO.php

```diff
@@ -149,7 +149,7 @@ protected function getFirstPrimaryKey(): string {
    */
   public function __construct() {
     $this->initialize();
-    $this->__table = $this->getTableName();
+    $this->__table = $this::getLocaleTableName();
```

Comment:
This is the one place where we *do* want to pass in the localized name because native DAO object queries don't go through `CRM_Core_I18n_Schema::rewriteQuery()`. I've skimmed the list of other callers to `::getTableName()` and I'm pretty sure this is the only one where the localized name is desirable (and without this change there were massive test failures, so I can confirm this is tested).